cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -70,19 +72,24 @@ class BufferExpander
     }
 
     BufferExpander(std::shared_ptr<ConcreteBuffer> const & buf, bool clone, ctor_passkey const &)
-        : m_concrete_buffer(clone ? buf->clone() : buf)
-        , m_begin(m_concrete_buffer->data())
-        , m_end(m_begin + m_concrete_buffer->size())
-        , m_end_cap(m_begin + m_concrete_buffer->size())
+        : BufferBase<BufferExpander>(nullptr, nullptr) //  initialize m_begin and m_end, but since we don't have the data yet, we set them to nullptr
```

Comment:
Don't delegate to the parent constructor setting `nullptr`.  The pointers will be overridden and explicitly initializing here wastes cycles.

Please review delegates and remove unnecessary ones.

It will be good to add a comment line explaining why not delegate.

Reply:
fixed.